src/components/WalletBalanceList/index.tsx

```diff
@@ -234,16 +268,119 @@ export const WalletBalanceList: React.SFC<WalletBalanceListProps> = ({
         })}
-    </div>
+      </div>
+    );
+  }
+
+  private renderAssetAddressModal = () => {
+    const QR_SIZE = 240;
+
+    return (
+      <div className="asset-address-modal">
+        {this.assetStore.selectedAsset &&
+        this.assetStore.selectedAsset.addressExtension ? (
+          <div>
+            <div className="asset-address-label">
+              {this.assetStore.selectedAsset.addressBase}
+            </div>
+            <hr />
+            <QRCode
+              size={QR_SIZE}
+              value={this.assetStore.selectedAsset.addressBase}
+            />
+            <div className="asset-address-label">Address</div>
+            <hr />
+            <div className="asset-address-label">
+              {this.assetStore.selectedAsset.addressExtension}
+            </div>
+            <hr />
+            <QRCode
+              size={QR_SIZE}
+              value={this.assetStore.selectedAsset.addressExtension}
+            />
+            <div className="asset-address-label">Tag</div>
+          </div>
+        ) : this.assetStore.selectedAsset &&
+        this.assetStore.selectedAsset.address ? (
+          <div>
+            <div className="asset-address-label">
+              {this.assetStore.selectedAsset.address}
+            </div>
+            <hr />
+            <QRCode
+              size={QR_SIZE}
+              value={this.assetStore.selectedAsset.address}
+            />
+          </div>
+        ) : (
+          <Spinner />
+        )}
+      </div>
+    );
+  };
+
+  private renderDepositMenuItem = (
+    label: string,
+    route: string,
+    iconUrl: string
+  ) => (
+    <DropdownListItem
+      key={label}
+      className={classnames({
+        disabled: !this.profileStore.isKycPassed
+      })}
+    >
+      {this.profileStore.isKycPassed ? (
+        <Link to={route}>
+          <img className="icon" src={iconUrl} />
+          {label}
+        </Link>
+      ) : (
+        <a>
+          <img className="icon" src={iconUrl} />
+          {label}
+        </a>
+      )}
+    </DropdownListItem>
   );
-};
-
-export default inject(({rootStore}: {rootStore: RootStore}) => ({
-  assetsAvailableForCreditCardDeposit:
-    rootStore.assetStore.assetsAvailableForCreditCardDeposit,
-  assetsAvailableForCryptoDeposit:
-    rootStore.assetStore.assetsAvailableForCryptoDeposit,
-  assetsAvailableForSwiftDeposit:
-    rootStore.assetStore.assetsAvailableForSwiftDeposit,
-  isKycPassed: rootStore.profileStore.isKycPassed
-}))(observer(WalletBalanceList));
+
+  private handleQrMouseOver = (assetId: string) => {
+    const asset = this.assetStore!.getById(assetId);
+    if (asset && !asset.address) {
+      this.assetStore!.fetchAddress(assetId);
+    }
+  };
+
+  @action
+  private handleQrClick = (assetId: string) => {
+    this.uiStore.showAssetAddressModal = true;
+    this.assetStore.selectedAsset = this.assetStore!.getById(assetId);
+  };
+
+  @action
+  private handleCloseAssetAddressModal = () => {
```

Comment:
I think it make sense to start adding modal windows that will work without uiStore OR using some high level functions that will show and hide that modal dialog without manually changing some boolean variable

Reply:
Why? For me this is exaclty what uiStore should handle) It's responsible for shadows, blur, etc if modal is active. https://mobx.js.org/best/store.html

Reply:
But in case you have 100 modal windows you will need to store 100 variables with getters and setters. I'm trying to make code more usable and for me some additional interface for modals and, probably, some modal store, will be better option.

Reply:
I got it, let's introduce this with the next modal?)

Reply:
As you wish